compatibility_lib/compatibility_lib/compatibility_store.py

```diff
@@ -192,36 +191,28 @@ def _compatibility_status_to_release_time_rows(
         rows = []
 
         for pkg, version_info in dependency_info.items():
-            row = {
-                'install_name': install_name,
-                'dep_name': pkg,
-            }
-            row.update(version_info)
-            row['timestamp'] = row.pop('current_time')
+            row = (install_name,
+                   pkg,
+                   version_info['installed_version'],
+                   version_info['installed_version_time'],
+                   version_info['latest_version'],
+                   version_info['latest_version_time'],
+                   version_info['is_latest'],
+                   version_info['current_time'])
             rows.append(row)
 
         return rows
 
-    @staticmethod
-    def _filter_older_versions(crs: Iterable[CompatibilityResult]) \
-            -> Iterable[CompatibilityResult]:
-        """Remove old versions of CompatibilityResults from the given list."""
-
-        def key_func(cr):
-            return frozenset(cr.packages), cr.python_major_version
-
-        filtered_results = []
-        crs = sorted(crs, key=key_func)
-        for _, results in itertools.groupby(crs, key_func):
-            filtered_results.append(max(results, key=lambda cr: cr.timestamp))
-        return filtered_results
-
     def get_packages(self) -> Iterable[package.Package]:
         """Returns all packages tracked by the system."""
-        query = 'SELECT DISTINCT install_name FROM {}'.format(
-            self._self_table_id)
-        query_job = self._client.query(query)
-        for row in query_job:
+        query = 'SELECT DISTINCT install_name FROM self_compatibility_status'
+
+        with closing(self.connect()) as conn:
+            with closing(conn.cursor()) as cursor:
```

Comment:
I don't think that you have to close the cursor. But you can if you want ;-)

Reply:
Keep it for now.